db/migrate/20240312122903_make_stripe_payment_intent_generic.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class MakeStripePaymentIntentGeneric < ActiveRecord::Migration[7.1]
+  def change
+    change_table :stripe_payment_intents do |t|
+      t.rename :user_id, :initiated_by_id # Make the purpose of the user_id column more clear
+      t.rename :confirmed_by_id, :confirmation_source_id
+      t.rename :confirmed_by_type, :confirmation_source_type
+      t.rename :canceled_by_id, :cancellation_source_id
+      t.rename :canceled_by_type, :cancellation_source_type
+
+      t.string :wca_status
+      t.string :payment_record_type
+      t.integer :payment_record_id
+    end
+
+    rename_table :stripe_payment_intents, :payment_intents
+    rename_column :stripe_records, :status, :stripe_status
+    rename_column :stripe_records, :api_type, :record_type
+    rename_column :paypal_records, :status, :paypal_status
+
+    reversible do |direction|
+      direction.up do
+        RegistrationPayment.update_all(receipt_type: 'StripeRecord')
+
+        PaymentIntent.update_all(payment_record_type: 'StripeRecord')
+
+        PaymentIntent.find_each do |intent|
+          intent.assign_attributes(payment_record_id: intent.stripe_record_id)
+          intent.set_wca_status
+          intent.save
+        end
+      end
+
+      direction.down do
+        RegistrationPayment.update_all(receipt_type: 'StripeTransaction')
+
+        PaymentIntent.find_each do |intent|
+          intent.assign_attributes(stripe_record_id: intent.payment_record_id)
+          intent.save(validate: false)
+        end
+      end
+    end
+
+    remove_column :payment_intents, :stripe_record_id, :integer
```

Comment:
I think it is sufficient if you _rename_ this column (and drop the `t.integer` column creation above).

Reply:
That's what I was doing initially, but it doesn't reorder them in the database - so you end up the record_id being the 3rd column, and record_type aaaaall the way at the end - which is (1) bad devex for developers who know there is a polymorphic association, and (2) doesn't make it obvious to developers who don't know there's a polymorphic association that one exists

Totally happy to rename if there's a way to reorder the SQL columns though

Reply:
In the statement that creates the column, add an `after: :existing_column_name` argument :wink: 

`t.string :cool_new_stuff, after: :existing_stuff_at_beginning_of_table`